storage/replica.go

```diff
@@ -911,16 +1061,61 @@ func (r *Replica) applyRaftCommandInBatch(ctx context.Context, index uint64, ori
 		}
 	}
 
-	// Execute the command.
-	reply, intents, rErr := r.executeCmd(batch, ms, args)
+	bReply := &proto.BatchResponse{}
+	bReply.Timestamp = bArgs.Timestamp
+	err, cache := func() (error, bool) {
+		for _, union := range bArgs.Requests {
+			args := union.GetValue().(proto.Request)
+
+			if lease := r.getLease(); args.Method() != proto.LeaderLease &&
```

Comment:
going to add a comment but leave it there for now. It's slightly more correct since we still play around with individual timestamps on the requests, so the batch header isn't authoritative. This is slated to be cleaned up #2155.